lightning/src/ln/interactivetxs.rs

```diff
@@ -2255,173 +2133,163 @@ mod tests {
-		OutputOwned::Single(TxOut {
-			value: Amount::from_sat(value),
-			script_pubkey: generate_p2sh_script_pubkey(),
-		})
+	fn generate_non_witness_output(value: u64) -> TxOut {
+		TxOut { value: Amount::from_sat(value), script_pubkey: generate_p2sh_script_pubkey() }
 	}
 
 	#[test]
 	fn test_interactive_tx_constructor() {
-		do_test_interactive_tx_constructor(TestSession {
-			description: "No contributions",
-			inputs_a: vec![],
-			outputs_a: vec![],
-			inputs_b: vec![],
-			outputs_b: vec![],
-			expect_error: Some((AbortReason::MissingFundingOutput, ErrorCulprit::NodeA)),
-			a_expected_remote_shared_output: None,
-			b_expected_remote_shared_output: None,
-		});
 		do_test_interactive_tx_constructor(TestSession {
 			description: "Single contribution, no initiator inputs",
 			inputs_a: vec![],
-			outputs_a: generate_output(&TestOutput::P2WPKH(1_000_000)),
-			inputs_b: vec![],
-			outputs_b: vec![],
-			expect_error: Some((AbortReason::OutputsValueExceedsInputsValue, ErrorCulprit::NodeA)),
-			a_expected_remote_shared_output: None,
-			b_expected_remote_shared_output: Some((generate_p2wpkh_script_pubkey(), 0)),
-		});
-		do_test_interactive_tx_constructor(TestSession {
-			description: "Single contribution, no initiator outputs",
-			inputs_a: generate_inputs(&[TestOutput::P2WPKH(1_000_000)]),
+			shared_output_a: generate_funding_txout(1_000_000, 1_000_000),
 			outputs_a: vec![],
 			inputs_b: vec![],
+			shared_output_b: generate_funding_txout(1_000_000, 0),
 			outputs_b: vec![],
-			expect_error: Some((AbortReason::MissingFundingOutput, ErrorCulprit::NodeA)),
-			a_expected_remote_shared_output: None,
-			b_expected_remote_shared_output: None,
+			expect_error: Some((AbortReason::OutputsValueExceedsInputsValue, ErrorCulprit::NodeA)),
 		});
+
 		do_test_interactive_tx_constructor(TestSession {
 			description: "Single contribution, no fees",
 			inputs_a: generate_inputs(&[TestOutput::P2WPKH(1_000_000)]),
-			outputs_a: generate_output(&TestOutput::P2WPKH(1_000_000)),
+			shared_output_a: generate_funding_txout(1_000_000, 1_000_000),
+			outputs_a: vec![],
 			inputs_b: vec![],
+			shared_output_b: generate_funding_txout(1_000_000, 0),
 			outputs_b: vec![],
 			expect_error: Some((AbortReason::InsufficientFees, ErrorCulprit::NodeA)),
-			a_expected_remote_shared_output: None,
-			b_expected_remote_shared_output: Some((generate_p2wpkh_script_pubkey(), 0)),
 		});
 		let p2wpkh_fee = fee_for_weight(TEST_FEERATE_SATS_PER_KW, P2WPKH_INPUT_WEIGHT_LOWER_BOUND);
 		let outputs_fee = fee_for_weight(
 			TEST_FEERATE_SATS_PER_KW,
-			get_output_weight(&generate_p2wpkh_script_pubkey()).to_wu(),
+			get_output_weight(&generate_p2wsh_script_pubkey()).to_wu(),
 		);
 		let tx_common_fields_fee =
 			fee_for_weight(TEST_FEERATE_SATS_PER_KW, TX_COMMON_FIELDS_WEIGHT);
 
 		let amount_adjusted_with_p2wpkh_fee =
-			1_000_000 - p2wpkh_fee - outputs_fee - tx_common_fields_fee;
+			1_000_000 - p2wpkh_fee - outputs_fee - tx_common_fields_fee + 1;
```

Comment:
It isn't clear to me why we add one here and also whenever we use this variable below.